niworkflows/interfaces/cifti.py

```diff
@@ -439,7 +440,7 @@ def _create_cifti_image(
     matrix.append(geometry_map)
     matrix.metadata = ci.Cifti2MetaData(meta)
     hdr = ci.Cifti2Header(matrix)
-    img = ci.Cifti2Image(bm_ts, hdr)
+    img = ci.Cifti2Image(dataobj=bm_ts, header=hdr, nifti_header=bold_img.header)
```

Comment:
You probably don't want to reused the NIfTI header. All you want to do is set the dtype.

```suggestion
    img = ci.Cifti2Image(dataobj=bm_ts, header=hdr)
    img.set_data_dtype(bold_img.get_data_dtype())
```

I think you could replace the whole PR with this. Maybe changing L349 to:

```Diff
-    bm_ts = np.empty((timepoints, 0))
+    bm_ts = np.empty((timepoints, 0), dtype=np.float32)
```

to save a little memory.